exporter/stackdriverexporter/stackdriver.go

```diff
@@ -115,20 +116,27 @@ func (se *stackdriverExporter) pushMetricsData(ctx context.Context, md consumerd
 	return se.exporter.PushMetricsProto(ctx, md.Node, md.Resource, md.Metrics)
 }
 
-// TODO(songya): add an interface PushSpanProto to Stackdriver exporter and remove this method
```

Comment:
I would leave the TODO as is. This is because PushSpanProto should have been implemented. In fact you did implement that in your first commit in SD exporter. However it converted spanProto to SpanData using translator defined in Otel-Collector which pulled in unnecessary dependency on otel-collector. But instead of converting spanProto to SpanData and then converting SpanData to [Stackdriver span](https://godoc.org/google.golang.org/genproto/googleapis/devtools/cloudtrace/v2#Span) it should have directly converted spanProto to Stackdriver Span (similar to [protoFromSpanData](https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/ab68e2a40809b13b9e06ac2135d2549a6a984d62/trace_proto.go#L47))

Reply:
I'm confused now, what should the api signatures look like?  Do you want a PushTraceProto in the exporter that takes straight OC trace protos, or the span data?  That means we'll have two almost identical methods in the exporter package, one for SpanData, and the other tracepb.Span

Or I can convert it directly in here, which will involve, yet another duplication of that logic.

Reply:
I'm confused now, what should the api signatures look like?  Do you want a PushTraceProto in the exporter that takes straight OC trace protos, or the span data?  That means we'll have two almost identical methods in the exporter package, one for SpanData, and the other tracepb.Span

Or I can convert it directly in here, which will involve, yet another duplication of that logic.

Reply:
The preference would be to have PushTraceProto in the exporter that takes OC trace protos. Then translate TraceProto to stackdriver span directly. Yes, there would be yet another function to convert but only one conversion will be used at any given time. Right now it is
TraceProto -> SpanData -> Stackdriver Span.

With the new api it would be
TraceProto -> Stackdriver Span.

Since TraceProto will be replaced with new OTLP, it is probably not worth making any more changes.

Reply:
ok I'll close the PR then, but just as an FYI we (shopify) are currently dropping millions of spans to stackdriver because of this issue combined with write api request quotas.

Reply:
Sorry for the confusion here. I didn't mean to imply to close this PR. This PR can still be merged to solve your problem. All I am saying is that PushTraceProto would have been a better choice as it would have avoided one data conversion.